File: TMK-Client/windowservice.go
package main

import "github.com/wailsapp/wails/v3/pkg/application"

var (
	mainWindow     application.Window
	subtitleWindow application.Window
)

type WindowService struct{}

func NewWindowService() *WindowService {
	return &WindowService{}
}

func (s *WindowService) ShowMain() {
	if mainWindow != nil {
		mainWindow.Show().Focus()
	}
}

func (s *WindowService) HideMain() {
	if mainWindow != nil {
		mainWindow.Hide()
	}
}

func (s *WindowService) ShowSubtitle() {
	setSubtitleVisible(true)
}

func (s *WindowService) HideSubtitle() {
	setSubtitleVisible(false)
}

func (s *WindowService) ToggleSubtitle() bool {
	if subtitleWindow == nil {
		return false
	}
	if subtitleWindow.IsVisible() {
		return setSubtitleVisible(false)
	}
	return setSubtitleVisible(true)
}

func setSubtitleVisible(visible bool) bool {
	if subtitleWindow == nil {
		return false
	}
	if visible {
		subtitleWindow.SetAlwaysOnTop(true).Show()
	} else {
		subtitleWindow.Hide()
	}
	application.Get().Event.Emit("subtitle-visibility-changed", visible)
	return visible
}

File: TMK-Client/frontend/bindings/changeme/windowservice.ts
// This file follows the generated Wails binding shape.

import { Call as $Call, CancellablePromise as $CancellablePromise } from "@wailsio/runtime";

export function ShowMain(): $CancellablePromise<void> {
    return $Call.ByName("main.WindowService.ShowMain");
}

export function HideMain(): $CancellablePromise<void> {
    return $Call.ByName("main.WindowService.HideMain");
}

export function ShowSubtitle(): $CancellablePromise<void> {
    return $Call.ByName("main.WindowService.ShowSubtitle");
}

export function HideSubtitle(): $CancellablePromise<void> {
    return $Call.ByName("main.WindowService.HideSubtitle");
}

export function ToggleSubtitle(): $CancellablePromise<boolean> {
    return $Call.ByName("main.WindowService.ToggleSubtitle");
}
